docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-self-hosted.mdx

```diff
@@ -258,6 +258,47 @@ $ tsh db logout oracle
 $ tsh db logout
 ```
 
+## Troubleshooting
+
+
+### Error on client connection
+
+If you are getting an error similar to the one below, that is likely a failure to connect
+to the TCPS service from the Teleport Database Service.
+
+```code
+$ tsh db connect --db-name=example --db-user=alice oracle
+SQLcl: Release 24.4 Production on Fri Feb 28 13:00:42 2025
+
+Copyright (c) 1982, 2025, Oracle.  All rights reserved.
+
+Connection failed
+  USER          =
+  URL           = jdbc:oracle:thin:@localhost:52019/example
+  Error Message = ORA-17800: Got minus one from a read call. (CONNECTION_ID=1234==)
+https://docs.oracle.com/error-help/db/ora-17800/
+
+ERROR: exit status 1
+```
+
+To resolve first check that `tcps` port is listening with the expected certificate with `lsnrctl status`.
+
+Run `openssl s_client -connect localhost:2484` on the Oracle host to validate the service. As part of the output you should see
+a certificate chain with the Oracle host and Teleport cluster.
+
+```bash
+....
+Certificate chain
+ 0 s:/CN=db.example.com
+   i:/O=example.teleport.sh/CN=example.teleport.sh/serialNumber=290149608632615397455959888366851033338
+ 1 s:/O=example.teleport.sh/CN=example.teleport.sh/serialNumber=290149608632615397455959888366851033338
+   i:/O=example.teleport.sh/CN=example.teleport.sh/serialNumber=290149608632615397455959888366851033338
+```
+
+If that chain is not available then confirm that the configuration matches the wallet paths. You may 
+need to recreate the wallet to resolve.  Once you have updated the configuration run the `openssl s_client -connect localhost:2484`
```

Comment:
What do they need to do to recreate the wallet? Do we just expect that the reader knows how to do this? Do we have other docs we can link to?

Reply:
The wallet creation is simply a section in this very same page (search for "Use the orapki tool on your Oracle server to set up an Oracle wallet"). Assuming this is a common stumbling block, I think it would be beneficial to expand the existing docs with the "verification" step as outline above: use `openssl` to get the reported certificate chains. It should fit right in the step about configuring TCPS listener, which would make the context for `lsnrctl status` command obvious.